### PR TITLE
Add  `empty-brace-spaces` rule

### LINKED ISSUE #900: Rule proposal: `empty-brace-space`
This rule would enforce spaces between braces in empty anonymous functions and try/catch braces. It is important because it is a syntax style I [need to follow](https://github.com/sindresorhus/internal-ip/pull/36#discussion_r527539515) but isn't automatically caught by xo already.

## Fail

```js
try { } catch { }

() => { }
```

## Pass

```js
try {} catch {}

() => {}
```

### PR DESCRIPTION
**Pull request by fisker**
I renamed `empty-brace-space` to `empty-brace-spaces`, consistent with `no-console-spaces` rule name.

Fixes #900

### CODE REVIEW

File: rules/empty-brace-spaces.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
+'use strict';
+const getDocumentationUrl = require('./utils/get-documentation-url');
+
+const MESSAGE_ID = 'empty-brace-spaces';
+const messages = {
+	[MESSAGE_ID]: 'Do not add spaces between braces.'
+};
+
+const selector = `:matches(${
```
Comment (Richienb) [inline rules/empty-brace-spaces.js:9]: Isn't it easier to just do this?

```js
const selector = ":matches(BlockStatement[body.length=0], ClassBody[body.length=0], ObjectExpression[properties.length=0])"
```

File: test/empty-brace-spaces.js [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+	'try {} catch(foo){/* */}',
+	'try {} catch(foo){} finally {/* */}',
+	'do {/* */} while (foo)',
+	'while (foo){/* */}',
+	'foo = () => {/* */}',
+	'foo = function (){/* */}',
+	'function foo(){/* */}',
+	'foo = {/* */}',
+	'class Foo {bar() {/* */}}',
+	'foo = class {bar() {/* */}}'
```
Comment (Richienb) [inline test/empty-brace-spaces.js:27]: We also need to test for `() => {}`.

File: rules/empty-brace-spaces.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+		'ClassBody[body.length=0]',
+		'ObjectExpression[properties.length=0]'
+	].join(', ')
+})`;
+
+const create = context => {
+	const sourceCode = context.getSourceCode();
+	return {
+		[selector](node) {
+			let [start, end] = node.range;
```
Comment (sindresorhus) [inline rules/empty-brace-spaces.js:21]: We should not use the unofficial properties `start` and `end`.

File: rules/empty-brace-spaces.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+					end: sourceCode.getLocFromIndex(end)
+				},
+				messageId: MESSAGE_ID,
+				fix: fixer => fixer.replaceTextRange([start, end], '')
+			});
+		}
+	};
+};
+
+const schema = [];
```
Comment (sindresorhus) [inline rules/empty-brace-spaces.js:41]: I don't see the point of adding empty schema.

**Review by sindresorhus — CHANGES_REQUESTED** [review]

File: rules/empty-brace-spaces.js (hunk 1 above)
Comment (fisker) [reply rules/empty-brace-spaces.js:9]: Yes, same. But I prefer current one, it's more readable add diff friendly, if we want add more.

```diff
const selector = `:matches(${
	[
		'BlockStatement[body.length=0]',
		'ClassBody[body.length=0]',
+		'ImportNamespaceSpecifier[assertions.length=0]',
		'ObjectExpression[properties.length=0]'
	].join(', ')
})`;
```

VS

```diff
- const selector = ":matches(BlockStatement[body.length=0], ClassBody[body.length=0], ObjectExpression[properties.length=0])"
+ const selector = ":matches(BlockStatement[body.length=0], ClassBody[body.length=0], ImportNamespaceSpecifier[assertions.length=0], ObjectExpression[properties.length=0])"

File: rules/empty-brace-spaces.js (hunk 3 above)
Comment (fisker) [reply rules/empty-brace-spaces.js:21]: That's destructuring `.range`  :smile:

Comment (sindresorhus) [reply rules/empty-brace-spaces.js:21]: Oh... I missed that.

File: test/empty-brace-spaces.js (hunk 2 above)
Comment (fisker) [reply test/empty-brace-spaces.js:27]: It's already tested https://github.com/sindresorhus/eslint-plugin-unicorn/pull/933/files#diff-049935872bb22d7098f0c8cbe3b5319cf79af03b3769b1387f3d3fa20cb8a546R22

File: rules/empty-brace-spaces.js (hunk 4 above)
Comment (fisker) [reply rules/empty-brace-spaces.js:41]: It's in template, forgot to remove.

BTW: Do you know `eslint-plugin-eslint-plugin` not working on our rules? And actually this is [enforced by it](https://github.com/not-an-aardvark/eslint-plugin-eslint-plugin/blob/master/docs/rules/require-meta-schema.md).

Comment (sindresorhus) [reply rules/empty-brace-spaces.js]: Not sure why it didn't catch this. It has caught issues in the past though.

**Pull request merged**